app/routers/api/auth.py

```
@@ -244,3 +255,354 @@ def generate_api_token(
+    # Extract roles/groups
+    roles_claim = instance_config_service.get_oidc_roles_claim(db)
+    oidc_roles = userinfo.get(roles_claim)
+
+    # Try decoding id_token as fallback if roles not in userinfo
+    if not oidc_roles and "id_token" in tokens:
+        try:
+            from jose import jwt
+
+            id_token_payload = jwt.decode(tokens["id_token"], "", options={"verify_signature": False})
+            oidc_roles = id_token_payload.get(roles_claim)
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)